src/GraphServiceClient.php

```diff
@@ -22,11 +25,35 @@
 class GraphServiceClient extends Generated\BaseGraphClient
 {
     /**
+     * @var RequestAdapter|null
+     */
+    private static ?RequestAdapter $customRequestAdapter = null;
```

Comment:
I don't think we want this to be static. In the case someone needs to support a scenario across different graph supported clouds, we will need to support having one GraphServiceClient per cloud. They cannot share RequestAdapter as each GraphServiceClient instance would need to target a different base url.

If this is the case, then we the constructor always needs RequestAdapter, so we don't need the else on line 40.

Reply:
To change the default national cloud, the developer would have to use `createWithRequestAdapter()` which overwrites the custom/temp request adapter variable.